presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkShuffleStats.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.classloader_interface;
+
+import java.io.Serializable;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.spark_project.guava.base.Preconditions.checkArgument;
+
+public class PrestoSparkShuffleStats
+        implements Serializable
+{
+    private final int fragmentId;
+    private final int taskId;
+    private final Operation operation;
+    private final long processedRows;
+    private final long processedBytes;
+    private final long elapsedWallTimeMills;
+
+    public PrestoSparkShuffleStats(
+            int fragmentId,
+            int taskId,
+            Operation operation,
+            long processedRows,
+            long processedBytes,
+            long elapsedWallTimeMills)
+    {
+        this.fragmentId = fragmentId;
+        this.taskId = taskId;
+        this.operation = requireNonNull(operation, "operation is null");
+        checkArgument(processedRows >= 0, "processedRows must be greater than or equal to zero: %s", processedRows);
+        this.processedRows = processedRows;
+        checkArgument(processedBytes >= 0, "processedBytes must be greater than or equal to zero: %s", processedBytes);
+        this.processedBytes = processedBytes;
+        checkArgument(elapsedWallTimeMills >= 0, "elapsedWallTimeMills must be greater than or equal to zero: %s", elapsedWallTimeMills);
+        this.elapsedWallTimeMills = elapsedWallTimeMills;
+    }
+
+    public int getFragmentId()
+    {
+        return fragmentId;
+    }
+
+    public int getTaskId()
+    {
+        return taskId;
+    }
+
+    public Operation getOperation()
+    {
+        return operation;
+    }
+
+    public long getProcessedRows()
+    {
+        return processedRows;
+    }
+
+    public long getProcessedBytes()
+    {
+        return processedBytes;
+    }
+
+    public long getElapsedWallTimeMills()
+    {
+        return elapsedWallTimeMills;
+    }
+
+    public enum Operation
+    {
+        READ,
+        WRITE
+    }
+
+    @Override
+    public String toString()
+    {
+        // readable summary to be displayed at the Spark web interface
+        return format(
+                "%s.%s:%s:%sM:%sMB:%smin",
+                fragmentId,
+                taskId,
+                operation.toString().charAt(0),
```

Comment:
I would prefer just `operation.toString()`

Reply:
No worry, spark accumulator will list all taskIds up to thousands, it is better to keep it short.